docs/fee_design.md

```diff
@@ -0,0 +1,169 @@
+
+```
+        {
+                txType: xxx,
+                inputs: [],
+                outputs: [],
+                nonce: xxx,
+        }
+```
+
+The list of inputs for fee transactions is empty, and the list of outputs only contains one output.
+
+This output follows the fungible token output format, which can be spent as an input for a Payment transaction.
+
+```
+        {
+                outputType: xxx,
+                amount: xxx,
+                outputGuard: xxx,
+                token: xxx,
+        }
+```
+To represent fees for multiple tokens, one would need to generate multiple fee transactions.
+
+The last field, `nonce`, would be computed via `hash(blockNum, token)`. `blockNum` is the block number that the fee transaction is mined to. `token` is the token that is claimed in the fee transaction output. This combination promises the `nonce` to be unique, and thus promises the fee transaction to be unique.
+
+Since this fee transaction would have a specific transaction type (and also its outputs would have unique output types), we don’t need to worry about other transaction types that use the same mechanism (block number + token) to ensure uniqueness. Even if another transaction type, for instance, collides with the same outputs in the same block, they would end up with different transaction hash due to the transaction type difference. Transaction uniqueness is granted.
+
+### Fee transaction type extension
+
+The current Plasma Framework design is immutable on the ability of spending an output type in another transaction type once the contracts have been deployed. So let's say we have payment v1 that can spend the fee claiming output. When we extend the framework with payment v2, we would need another fee output type that is able to be spent in payment v2. As a result, we would need fee transaction type 2 as well during the extension. (As only new transaction type can create new output type)
+
+Given this, the block verification should limit the block to be existing with a singe fee transaction type per block for simplicity. No matter which fee transaction types (1 or 2), it should calculate the fee balance of all transaction within the block. The only difference is how the output can be spent.
+
+Child chain and watcher could even further deprecate old fee transaction types afterward if not needed anymore. The logic change could be done by upgrading Child Chain service and watcher together.
+
+## Adding the fee exit feature to the Plasma Framework
+
+This section will discuss how we can add the fee exit feature to our Plasma Framework using the chosen design, assuming we are launching the network without fee exit feature at the beginning.
+
+Since the network would first be running with a Payment transaction type which does not support having fee transaction type as input, to enable spending fee transaction into Payment transaction, we would need a Payment transaction v2 for that.
```

Comment:
```suggestion
Since the network would first be running with a payment transaction type which does not support having fee transaction type as input, to enable spending fee transaction into Payment transaction, we would need a Payment transaction v2 for that.
```
Not sure about this one... this is re-occuring, so maybe just italic this and lowercase? idk